aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/AddVersionCommand.java

```diff
@@ -276,7 +282,26 @@ protected Integer processCallCommand() throws Exception {
                     log.info("[Debug options] Amt Profiling file download URL: {}",
                             sharedOptions.getFullServerRootUrl() + ApiEndpointHelper.getAmtProfilingDownloadUrl(applicationGuid));
                 }
-                log.info("Job completed successfully.");
+                log.info("{}: Job completed successfully.", builder.buildJobRequest().getJobType().getLabel());
+
+                //Check Fast-Scan workflow pre-conditions and trigger publish to imaging
+                ApplicationDto applicationDto = applicationService.getApplicationDetails(applicationGuid);
+                Set<String> statuses = EnumSet.of(VersionStatus.ANALYSIS_DATA_PREPARED, VersionStatus.IMAGING_PROCESSED,
+                                VersionStatus.SNAPSHOT_DONE, VersionStatus.FULLY_ANALYZED, VersionStatus.ANALYZED).stream()
+                        .map(VersionStatus::toString).collect(Collectors.toSet());
+                VersionDto versionDto = applicationDto.getVersion();
+                if (!statuses.contains(versionDto.getStatus().toString())) {
```

Comment:
```suggestion
                EnumSet<VersionStatus> statuses = EnumSet.of(VersionStatus.ANALYSIS_DATA_PREPARED, VersionStatus.IMAGING_PROCESSED, VersionStatus.SNAPSHOT_DONE, VersionStatus.FULLY_ANALYZED, VersionStatus.ANALYZED);
                VersionDto versionDto = applicationDto.getVersion();
                if (!statuses.contains(versionDto.getStatus())) {
```
To work with Enums, EnumSet is a bit better, and it avoid mapping enums to strings.

Reply:
yes